src/ssl.c

```diff
@@ -44374,35 +44374,26 @@ int wolfSSL_i2d_X509_NAME(WOLFSSL_X509_NAME* name, unsigned char** out)
         return BUFFER_E;
     }
 
-    /* check if using buffer passed in */
-    if (*out == NULL) {
-        *out = local = (unsigned char*)XMALLOC(totalBytes + idx, NULL,
-                DYNAMIC_TYPE_OPENSSL);
+    if (out != NULL) {
```

Comment:
Switch around if to make it clearer.

if (out == NULL) {
    ...
    return ...
}

if (*out == NULL) {
...